### ISSUE TITLE
Add method to created conditional writer that takes no config.

### ISSUE THREAD

**Issue by keith-turner** [apache/accumulo#680]
In AccumuloClient a method for creating conditional writers that takes no config could be added.  Doing this would probably require also adding client properties for the conditional writer.

**Comment by DomGarguilo** [comment]
Are you suggesting having a method that accepts a table name as its sole parameter?

> Doing this would probably require also adding client properties for the conditional writer.

What do you think would need to be added?

**Comment by ctubbsii** [comment]
> In AccumuloClient a method for creating conditional writers that takes no config could be added.

We could add an overloaded `createConditionalWriter(String tableName)` that just calls `createConditionalWriter(tableName, new ConditionalWriterConfig())`, but conditional writers seem used infrequently enough that it doesn't really warrant the overload, since this is already pretty easy to do.

> Doing this would probably require also adding client properties for the conditional writer.

It wouldn't *require* it, but we could make it so `new ConditionalWriterConfig()` gets default values from client properties. How common is it to have an application need to create many different ConditionalWriters with the same ConditionalWriterConfig, though? And, it seems convenient enough to be able to reuse a ConditionalWriterConfig instance and pass it as the second arg in those cases. I'm not sure the advantage to growing client properties over simply reusing the object.

**Comment by keith-turner** [comment]
> Are you suggesting having a method that accepts a table name as its sole parameter?

Yeah that is what I was thinking, like [createBatchWriter(String)](https://github.com/apache/accumulo/blob/af2d2d630094fc06e0ac1d6702a9a49d30ec1e94/core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java#L195), `createBatchScanner(String)`, `createScanner(String)`.  All of those method were added for 2.0. I had wanted to add one for the conditionalWriter before 2.0 to make it consistent, but I suppose it slipped through the cracks.

**Comment by ctubbsii** [comment]
Making it consistent is a good idea. I'm still not sure about changing the defaults via client properties though. That may or may not be worth it.

**Comment by DomGarguilo** [comment]
It does seem like it could clean up ConditionalWriterIT a bit seeing that `createConditionalWriter(tableName, new ConditionalWriterConfig()))` is used a good number of times.

**Comment by keith-turner** [comment]
>  I'm still not sure about changing the defaults via client properties though.

I was thinking the client prop behavior should be consistent w/ the other methods. Looking into it, the method w/o a config object is completely unrelated to client config override for createBatchWriter() methods.  For example the following will merge in client config for batch writer for either of the createBatchWriter methods.

https://github.com/apache/accumulo/blob/af2d2d630094fc06e0ac1d6702a9a49d30ec1e94/core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java#L535

I think when I write the ticket I was thinking they are related.  They are not related from an impl standpoint.  I think the only relation is more from a usability standpoint, in that the methods w/ no config obj exists to support only using client props.

**Comment by keith-turner** [comment]
> setting a config allows you to explicitly bypass those defaults
 
It also bypasses any client props.

**Comment by keith-turner** [comment]
> just expressing my skepticism that it would be useful.

In my opinion it would be useful.  Configuring via client props is so much better than configuring things in Java code when the situation supports it.  Maybe I can get to it for 2.1 if there is no interest.  May also be nice to look for any other thing that can be configured via java API in client and not by client props, not that everything needs a client prop but it may be nice to know what else is missing.

**Comment by keith-turner** [comment]
I opened #2115 after poking around in the existing code as a result of looking into this issue.

**Comment by ctubbsii** [comment]
> Configuring via client props is so much better than configuring things in Java code when the situation supports it.

You're right. I was actually thinking of the case where client properties is long-lived, and was skeptical that client properties for this would be appropriate for the lifetime of a long-lived client, or reusable for many short-lived clients using the same, unmodified, properties file.

However, your phasing here highlights another use case I had overlooked: short-lived client props for the purposes of executing a specific client. Using client props avoids a lot of boilerplate Java code for this case.

In short, my skepticism has dissipated. This is definitely a good idea.

**Comment by DomGarguilo** [comment]
I'm not sure I understand what is being proposed in terms of using client properties. Is it being suggested that properties be added to the client in order to create a `ConditionalWriterConfig`?

**Comment by DomGarguilo** [comment]
I would like to be assigned to this issue please.

**Comment by DomGarguilo** [comment]
I created #2116 in response to this ticket. @keith-turner @ctubbsii
